### ISSUE TITLE
Emulated Mouseclick only works when Viewport is in its original size - when not it jumps

### ISSUE THREAD

**Issue by Daniel-da-Silva** [godotengine/godot#44081]
Godot version: 3.2.3

MX Linux 19.3 always updated, i5-2520M, GLES 2.0 (but the problem also appeared in GLES 3.0)

Issue description:

To enable controller support for my whole game - which includes GUI buttons - I need to emulate mouse clicks. 
This works in the normal Viewport but when I toggle fullscreen the button is missed or can be clicked from weird places.
To check if it is the position of the mouse, I included some code that jumps the cursor to the position which the engine seems to think it is. This only happens when the GUI elements are 2D stretched but I need that (or is there another way?)

Steps to reproduce:
Hover over the button.
Press ESC
Enjoy the button animation
Press Enter (to toggle fullscreen)
Hover away from the button a little bit to the right
Click with the mouse -> no animation as expected
Press ESC -> mouse jumps, button is pressed

Minimal reproduction project:
[TestProjectMouseEmulation.zip](https://github.com/godotengine/godot/files/5642009/TestProjectMouseEmulation.zip)

```
extends Control

var mouse_pos : Vector2

func _physics_process(delta):

	if Input.is_action_just_pressed("ui_accept") and OS.window_fullscreen == false:
		OS.window_fullscreen = true

	elif Input.is_action_just_pressed("ui_accept") and OS.window_fullscreen == true:
		OS.window_fullscreen = false

	if Input.is_action_just_pressed("ui_cancel"):
		click_the_left_mouse_button()

func click_the_left_mouse_button():

	var evt = InputEventMouseButton.new()
	evt.button_index = BUTTON_LEFT
	evt.position = get_viewport().get_mouse_position()

	mouse_pos = get_viewport().get_mouse_position()
	Input.warp_mouse_position(mouse_pos)

	evt.pressed = true
	get_tree().input_event(evt)
	yield(get_tree().create_timer(0.5), "timeout")
	evt.pressed = false
	get_tree().input_event(evt)

```

**Comment by Calinou** [comment]
Duplicate of https://github.com/godotengine/godot/issues/29570.

**Comment by Daniel-da-Silva** [comment]
Oh okay, sorry, didn't saw that - will it be fixed soon or with 4.0?

**Comment by Calinou** [comment]
> Oh okay, sorry, didn't saw that - will it be fixed soon or with 4.0? And is there a way around that now? Thanks for your time

We don't know how to fix this bug, which means there is no set milestone for fixing this.

**Comment by Daniel-da-Silva** [comment]
Oh, that's a shame - thanks for the code - I'll test it :D

**Comment by Daniel-da-Silva** [comment]
It didn't work or I didn't implement properly (the latter is more likely) - but thanks for the idea with multiplying - this works almost perfect and should be more than enough (it jumps only a few pixels and only sometimes - I think this is due execution order of the code after I moved the mouse fast). 

Thanks again, this helped me tremendously!

Since it's not known how this bug originated: Would it be possible to add something like a checkable workaround option in the editor that adds something like a multiplier for the get_viewport().get_mouse_position()?
Like screen_multiplier = float(get_viewport().size.x/base_viewport.x) but in C++? (and not only for the keep option as I did)
Since mouse movement is fast and precise I think it would benefit from the performance gain (and maybe saves some headaches).
